File: android/app/src/main/kotlin/com/goalmaker/app/application/environment/BackendEnvironment.kt
package com.goalmaker.app.application.environment

/** Which Supabase project the app talks to: its API URL and publishable key. */
data class BackendEnvironment(
    val url: String,
    val publishableKey: String,
) {
    val isConfigured: Boolean get() = url.isNotBlank() && publishableKey.isNotBlank()
}

File: windows/src/GoalMaker.Core/Backend/BackendEnvironment.cs
namespace GoalMaker.Core.Backend;

/// <summary>Which Supabase project the app talks to: its API URL and publishable key.</summary>
public sealed record BackendEnvironment(string Url, string PublishableKey)
{
    public bool IsConfigured => !string.IsNullOrWhiteSpace(Url) && !string.IsNullOrWhiteSpace(PublishableKey);
}
